dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -807,6 +825,12 @@ private BigDecimal getUnsignedDecimal(int numDigits, int precision, int dataLen)
         return ret;
     }
 
+    private static BigInteger extractUnsigned8Val(long unsigned8)
+    {
+        return (BigInteger.valueOf((unsigned8 >> 32) & 0xffffffffL).shiftLeft(32))
```

Comment:
what are the expected bitlen and sign of resulting BigInt and how are we ensuring those?
Also, element comment would be helpful.

Reply:
Good question, will add some comments here